TR4 Sequence Switch Node

TrainWrack opened the pull request:
Adds Sequence Switch Node

For sequence switch place the volume covering all 3 switches. name the switches any prefix such as SWITCH_1; SWITCH_2; SWITCH_3

Name doors DOOR_1; DOOR_2; DOOR_3; DOOR_4; DOOR_5; DOOR_6

Add the words SWITCH and DOOR to the node.

Refer attached project. 
[TR4 Objects.zip](https://github.com/user-attachments/files/26555530/TR4.Objects.zip)

TrainWrack committed:
Add files via upload

TrainWrack committed:
Fix duplicate name

TrainWrack committed:
typos

TrainWrack committed:
Update Environment.lua

Lwmte committed:
Update Environment.lua

TrainWrack committed:
Add files via upload

TrainWrack committed:
Add files via upload

TrainWrack committed:
Update Environment.lua

TrainWrack committed:
Update Environment.lua

TrainWrack committed:
Update Environment.lua

TrainWrack committed:
Update Environment.lua

Lwmte committed:
Update Environment.lua

Lwmte committed:
Added a condition to prevent horizon crossfade into the same slot

Lwmte committed:
Removed unnecessary args

Lwmte committed:
Merge branch 'develop' into pr/1001

Lwmte committed:
Update Changes.txt

Lwmte committed:
Fixed two more mistakes

Lwmte committed:
Merge branch 'develop' into pr/1001

Lwmte committed:
Merge branch 'develop' into pr/1001

TrainWrack committed:
WIP

TrainWrack committed:
Finish Environment node

TrainWrack committed:
UpdateNodes

TrainWrack committed:
Cleanup Ammo counter nodes layout

Lwmte committed:
UI node catalog typo fixes

TrainWrack committed:
SimplifyKeypad Nodes

TrainWrack committed:
Merge branch 'develop' into develop

TrainWrack committed:
PathNodes

TrainWrack committed:
Merge branch 'MontyTRC89:develop' into develop

TrainWrack committed:
BurningFloorNode

TrainWrack committed:
Delete TombLib/TombLib/Catalogs/TEN Node Catalogs/Path.lua

TrainWrack committed:
Fix merge deltas

TrainWrack committed:
Merge remote-tracking branch 'upstream/develop' into BurningFloorNode

Stranger1992 commented on TombLib/TombLib/Catalogs/TEN Node Catalogs/Objects.lua:
```diff
@@ -0,0 +1,123 @@
+-- !Name "Burning Floor"
+-- !Section "Objects"
+-- !Description "Create a Burning Floor Object."
+-- !Arguments "NewLine, Moveables"
```
No tooltips

Stranger1992 commented on TombLib/TombLib/Catalogs/TEN Node Catalogs/Objects.lua:
```diff
@@ -0,0 +1,123 @@
+}
+
+local FIRE_OFFSETS =
+{   --displacement, rotation, size
```
Are these offsets relative to the OG Burning Floor object?

TrainWrack replied:
Yes.

Stranger1992 commented on TombLib/TombLib/Catalogs/TEN Node Catalogs/Objects.lua:
```diff
@@ -0,0 +1,123 @@
+local START_DELAY = 0.25
+local BURN_DURATION = 1024
```
I think the burning parameters should be included in the node itself. So the effect can be customised per burning floor.

TrainWrack replied:
I can certainly make it an option.

TrainWrack committed:
Addd SequenceSwitches, Scales, Elemental Puzzles

TrainWrack committed:
Finish Nodes

TrainWrack committed:
Remove Burning Floor and Elemental Puzzle code

Removed the Burning Floor and Elemental Puzzle functionalities, including related constants and functions.

Please use Native TEN versions

TrainWrack committed:
add sequence switches functionality
